src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -459,22 +459,40 @@ bool os::supports_sse() {
 }
 
 juint os::cpu_microcode_revision() {
+  // Note: this code runs on startup, and therefore should not be slow.
+
   juint result = 0;
-  char data[2048] = {0}; // lines should fit in 2K buf
+  char data[128] = {0}; // looking for short "microcode" line, should be enough
   size_t len = sizeof(data);
-  FILE *fp = os::fopen("/proc/cpuinfo", "r");
-  if (fp) {
-    while (!feof(fp)) {
+
+  // Attempt 1 (faster): Read the microcode version off the sysfs.
+  {
+    FILE *fp = os::fopen("/sys/devices/system/cpu/cpu0/microcode/version", "r");
+    if (fp) {
       if (fgets(data, len, fp)) {
-        if (strstr(data, "microcode") != NULL) {
-          char* rev = strchr(data, ':');
-          if (rev != NULL) sscanf(rev + 1, "%x", &result);
-          break;
+        sscanf(data, "%x", &result);
+      }
+      fclose(fp);
```

Comment:
The diff would be much simpler if you added a return here and so avoided the need for the if statement below.

Reply:
See new commit.

Reply:
Well, I don't see what you mean then, so I am open for clear suggestion how would you write it.